### PR TITLE
Add support for Swift Package Manager

### PR DESCRIPTION
**Pull request by simba909**
This PR adds support for Swift Package Manager to the project. I should point out that this is based off of the great stuff already done by @fabio-cerdeiral-ck in #3692, and by @sharplet in #3676. Big thanks to those guys for putting in the effort 👏 

Please let me know if there's anything that needs fixing!

#### Checklist
- [x] Updated CHANGELOG.md.

### CODE REVIEW

**Comment by simba909** [comment]
Not sure why the tests fail. I've run them on my machine, and there they fail on both `master` and this branch 🤔  @andersio Do you have any insights?

**Comment by andersio** [comment]
Hmm flaky test.

File: README.md [hunk 1]
```diff
@@ -114,6 +114,15 @@ ReactiveCocoa to your `Podfile`:
 pod 'ReactiveCocoa', '~> 9.0'
 ```
 
+#### Swift Package Manager
+
+If you use Swift Package Manager, simply add ReactiveCocoa as a dependency
+of your package in `Package.swift`:
+
+```
+.package(url: "https://github.com/ReactiveCocoa/ReactiveCocoa.git", branch: "master")
```
Comment (fabio-cerdeiral-ck) [inline README.md]: we should update the version next, then we can update these instructions to point to the right version

**Comment by NachoSoto** [comment]
Awesome! 🎉 
Could you add a test for `swift build` in `.travis.yml`?

**Comment by simba909** [comment]
@NachoSoto sure thing, will add it asap 👍🏻

**Comment by NachoSoto** [comment]
Build is failing.

**Comment by gcox** [comment]
@simba909 The build is failing because the `ReactiveCocoa.podspec` file needs to be tweaked to account for the new `ReactiveCocoaObjC/include` folder.

Changing [lines 18-19](https://github.com/simba909/ReactiveCocoa/blob/feature/add-swift-pm/ReactiveCocoa.podspec#L18) to the following fixes the `pod lib lint` errors:
```ruby
s.source_files = "ReactiveCocoa/*.{swift,h,m}", "ReactiveCocoa/Shared/*.{swift}", "ReactiveCocoaObjC/**/*.{h,m}"
s.public_header_files = "ReactiveCocoaObjC/include/ObjCRuntimeAliases.h"
```

**Comment by simba909** [comment]
Great, thanks @gcox ! Will add the change asap 👍

**Comment by NachoSoto** [comment]
👏🏻

**Comment by NachoSoto** [comment]
Is anyone actually able to use this from Xcode? I get tons of errors:
Screen Shot 2020-02-10 at 7 07 17 PM

**Comment by simba909** [comment]
@NachoSoto I just tried setting up a new empty project with ReactiveCocoa and it seems to work as expected (Xcode 11.4 beta 1, but it's also been working for me on earlier versions). Just to be completely sure, have you tried manually adding ReactiveSwift as a Swift package?

**Pull request merged**